### PR TITLE
feat(mac-helper): detect holds of a configured modifier set

### PR DESCRIPTION
**Pull request by alex-nork**
## Summary

- Adds `ModifierHoldDetector`: a pure state machine that distills the raw keyboard stream into holds of one configured set of modifiers.
- **Nothing reads it yet.** This is the first of a stack; the binding, the transport and the consumer land on top.

## Why a second detector rather than generalizing the first

`FnTapDetector` answers a different question, and the difference is not the modifier it watches. A tap is only known to be a tap once it has been released, so it emits `down` and `up` together at release. A microphone needs the opening edge while the key is still down, so a hold has to report at press and close later.

Keeping them apart means Fn keeps behaving exactly as it does, which matters because the single existing consumer treats the `down` edge as a toggle and discards the `up`.

## The rule worth reviewing

A hold that ends for any reason other than the set being released **stays ended**, and only a fresh press starts another. Holding the set and then pressing a key is someone else's shortcut passing through, and reopening the moment that key lifts would open a microphone in the middle of it.

This matters more than it would for Fn: Ctrl+Option is VoiceOver's own modifier, and macOS claims Ctrl+Option+F and friends, so the disqualified path is the common one rather than the edge case.

## Testing

`cd clients/macos/native/mac-helper && swift test` — 26 tests, no app build. Two of the eleven new ones are there because they failed first: the disqualified-hold latch was documented but not implemented, so the hold re-opened when the extra modifier lifted; and `cancel()` on an idle detector poisoned the next press. There is also a property test asserting every open is closed exactly once across all four exit paths, since the consumer's bookkeeping is a live microphone and an unmatched edge either strands it open or closes one that never opened.

### CODE REVIEW

File: clients/macos/native/mac-helper/Sources/MacHelperCore/ModifierHoldDetector.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                open = false
+                spent = targetHeld
+                return [.up]
+            }
+            return []
+        }
+
+        guard targetHeld else {
+            spent = false
+            return []
```
[the remark this answers is not in the document]
Comment (alex-nork) [reply clients/macos/native/mac-helper/Sources/MacHelperCore/ModifierHoldDetector.swift]: Addressed in 5ab6161a52
